domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt

```diff
@@ -2,25 +2,51 @@ package org.oppia.domain.topic
 
```

Comment:
Please also add tests to verify that randomness can result in different practice sessions being generated, and that the number of questions per training session is respected.

To simplify both of these tests, I suggest that you:
1. Move the number of questions per training session constant to a module; this requires tests to provide the value directly, and doing so allows you to specify a smaller number for the session sizes to simplify the tests
2. Seed the default Kotlin random number generator so that the tests are deterministic; you can use different seeds for different tests to ensure that the results change (though we will need enough questions to make sure that there are several configuration possibilities for the training sessions)

Reply:
The function that creates the question set is private so should I make it visible for testing and test that function?

Reply:
Prefer not doing that as it introduces a backdoor for testing purposes (prefer only testing via the public API).

I think we can test these behaviors with the public API by seeding the random number generator in a deterministic way, then verifying the correct assessment is started via the training controller.

Note that I think this comment meant to refer to QuestionTrainingControllerTest instead of TopicControllerTest.

Reply:
Okay got it, made these changes. We may have to mock out the QuestionAssessmentProgressController as well in the future once that is created